sapp/tests/warning_messages_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import pathlib
+import tempfile
+from contextlib import ExitStack
+from unittest import TestCase
+
+from sqlalchemy import Table
+
+from ..db import DB, DBType
+from ..models import WarningMessage, create as create_model
+from ..warning_messages import update_warning_messages, upsert_entry
+
+test_metadata = {
+    "codes": {
+        "1001": "Updated warning message from test",
+        "1002": "Updated warning message from test 2",
+    },
+    "codes2": {
+        "2001": "Updated warning message from another test",
+        "2002": "Updated warning message from another test 2",
+    },
+}
+
+
+class WarningMessagesTest(TestCase):
+    def setUp(self) -> None:
+        self.db = DB(DBType.MEMORY)
+        create_model(self.db)
+
+    def test_update_warning_messages(self) -> None:
+        with ExitStack() as stack:
+            temp = stack.enter_context(tempfile.NamedTemporaryFile(mode="w+"))
+            json.dump(test_metadata, temp)
+            temp.read()
```

Comment:
What's the point of the read here, if we are not using the result?

Reply:
If we don't call `.read()`, we are not able to use the file content of it later on the line number 41

Reply:
It looks like `temp.read()` reads the contents we dumped with `json.dump(test_metadata, temp)`. Without the call, `temp` will be empty when we call `path.Pathlib.read_text(temp.name)`. 

It looks like we might be able to avoid calling `temp.read()` if we set the mode during instantiation of `temp` to `'w+b'`, but then we cannot write a string to `temp`, we would have to encode `test_metadata` as byte object to be able to write it to `temp`. Would you prefer if we approach it this way instead @arthaud?

Reply:
I honestly still don't understand why we would need that `.read()`.
My only idea is that `.read()` flushes some buffer and the next read actually works? What if we use `temp.flush()` instead?
Also, not sure `w+b` will change anything.

Reply:
`temp.flush()` works as well, I can change it to that 😄